@xen-orchestra/backups/RemoteAdapter.js

```diff
@@ -688,6 +698,26 @@ class RemoteAdapter {
         return validator.apply(this, arguments)
       },
     })
+    if (expectedLength !== undefined) {
```

Comment:
Integrate all of this in the `validator` so that:
- the temp file is deleted on errors
- the temp file is renamed only after it's been checked